openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -374,6 +379,39 @@ def run(self):
         self._quantities = []
 
         optimizer = self.options['optimizer']
+
+        # Need to tell optimizer where to put its .out files
+        if self.options['output_dir'] is None:
+            self.options['output_dir'] = "."
+        elif self.options['output_dir'] == _DEFAULT_REPORTS_DIR:
+            problem = self._problem()
+            default_output_dir = pathlib.Path(get_reports_dir()).joinpath(problem._name)
+            pathlib.Path(default_output_dir).mkdir(parents=True, exist_ok=True)
+            self.options['output_dir'] = str(default_output_dir)
+        output_dir = self.options['output_dir']
+
+        if optimizer == 'ALPSO':  # Actually, this is the root of two files generated
```

Comment:
I think you could just use a dictionary like you do in the test and just pull the setting for the optimizer you need rather than have this extended if/then/else construct.

You might also consider moving this code down to around line 572 where it would be collocated with other opt_setting logic